sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryItemTests.cs

```diff
@@ -427,100 +433,92 @@ public void OTelResourceMetricTelemetryHasAllResourceAttributes()
-                Assert.Equal("bar", metricsData?.Properties["foo"]);
-            }
-            catch (Exception)
-            {
-                Environment.SetEnvironmentVariable(EnvironmentVariableConstants.EXPORT_RESOURCE_METRIC, null);
-            }
+            Assert.Equal("my-service", metricsData?.Properties[SemanticConventions.AttributeServiceName]);
+            Assert.Equal("my-namespace", metricsData?.Properties[SemanticConventions.AttributeServiceNamespace]);
+            Assert.Equal("my-instance", metricsData?.Properties[SemanticConventions.AttributeServiceInstance]);
+            Assert.Equal("my-deployment", metricsData?.Properties[SemanticConventions.AttributeK8sDeployment]);
+            Assert.Equal("my-pod", metricsData?.Properties[SemanticConventions.AttributeK8sPod]);
+            Assert.Equal("bar", metricsData?.Properties["foo"]);
         }
 
         [Fact]
         public void OTelResourceMetricTimesAreDifferent()
         {
-            try
-            {
-                Environment.SetEnvironmentVariable(EnvironmentVariableConstants.EXPORT_RESOURCE_METRIC, "true");
-                var instrumentationKey = "00000000-0000-0000-0000-000000000000";
+            var instrumentationKey = "00000000-0000-0000-0000-000000000000";
 
-                var testAttributes = new Dictionary<string, object>
+            var testAttributes = new Dictionary<string, object>
                 {
                     { "foo", "bar" }
                 };
 
-                using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
-                using var activity1 = activitySource.StartActivity(
-                    ActivityName,
-                    ActivityKind.Server,
-                    null,
-                    startTime: DateTime.UtcNow);
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            using var activity1 = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Server,
+                null,
+                startTime: DateTime.UtcNow);
+
+            Assert.NotNull(activity1);
 
-                Assert.NotNull(activity1);
+            var resource = ResourceBuilder.CreateEmpty().AddAttributes(testAttributes).Build();
+            var azMonResource = resource.CreateAzureMonitorResource(platform: GetMockPlatform("true"), instrumentationKey: instrumentationKey);
 
-                var resource = ResourceBuilder.CreateEmpty().AddAttributes(testAttributes).Build();
-                var azMonResource = resource.CreateAzureMonitorResource(instrumentationKey);
+            var telemetryItems = TraceHelper.OtelToAzureMonitorTrace(new Batch<Activity>(new Activity[] { activity1 }, 1), azMonResource, instrumentationKey, 1.0f);
+            var telemetryItem1 = telemetryItems.First(x => x.Name == "Metric"); // Collect the "_OTELRESOURCE_" metric.
```

Comment:
This is a fix for a test bug. Ping me offline, we can debug and see this test fail in the `main` branch.